wagtail/admin/views/generic/mixins.py

```diff
@@ -0,0 +1,35 @@
+from django.conf import settings
+from django.shortcuts import get_object_or_404
+
+from wagtail.models import Locale, TranslatableMixin
+
+
+class LocaleMixin:
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.locale = self.get_locale()
+
+    def get_translations_context(self):
```

Comment:
Or maybe just `get_translations(self)`?

Reply:
Should we rename this to ``get _locales`` and make it return a list/queryset of locales to display in the dropdown?

It feels like the formatting of the locale data for the template context belongs in ``get_context_data``?